src/main/java/org/peergos/net/APIHandler.java

```diff
@@ -0,0 +1,122 @@
+package org.peergos.net;
+
+import io.libp2p.core.Host;
+import io.libp2p.core.PeerId;
+import org.peergos.blockstore.Blockstore;
+import org.peergos.util.*;
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.*;
+import java.util.function.Function;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+
+public class APIHandler implements HttpHandler {
+	private static final Logger LOG = Logging.LOG();
+
+    private static final boolean LOGGING = true;
+
+    private final Blockstore store;
+    private final Host node;
+    private final String apiPrefix;
+
+    public static final String ID = "id";
+    public static final String VERSION = "version";
+
+    public APIHandler(Blockstore store,
+                      String apiPrefix,
+                      Host node) {
+        this.store = store;
+        this.apiPrefix = apiPrefix;
+        this.node = node;
+    }
+
+    @Override
+    public void handle(HttpExchange httpExchange) {
+        try {
+            if (!HttpUtil.allowedQuery(httpExchange)) {
+                httpExchange.sendResponseHeaders(403, 0);
+            } else {
+                handleCallToAPI(httpExchange);
+            }
+        } catch (Exception e) {
+            Throwable t = Exceptions.getRootCause(e);
+            LOG.severe("Error handling " + httpExchange.getRequestURI());
+            LOG.log(Level.WARNING, t.getMessage(), t);
+            HttpUtil.replyError(httpExchange, t);
+        }
+    }
+    public void handleCallToAPI(HttpExchange httpExchange) {
+
+        long t1 = System.currentTimeMillis();
+        String path = httpExchange.getRequestURI().getPath();
+        try {
+            if (! path.startsWith(apiPrefix))
+                throw new IllegalStateException("Unsupported api version, required: " + apiPrefix);
+            path = path.substring(apiPrefix.length());
+            // N.B. URI.getQuery() decodes the query string
+            Map<String, List<String>> params = HttpUtil.parseQuery(httpExchange.getRequestURI().getQuery());
+            List<String> args = params.get("arg");
+            Function<String, String> last = key -> params.get(key).get(params.get(key).size() - 1);
+
+            switch (path) {
+                case ID: { // https://docs.ipfs.tech/reference/kubo/rpc/#api-v0-id
+                    PeerId peerId = node.getPeerId();
+                    Map res = new HashMap<>();
+                    res.put("ID",  peerId.toBase58());
+                    replyJson(httpExchange, JSONParser.toString(res));
+                    break;
+                }
+                case VERSION: { // https://docs.ipfs.tech/reference/kubo/rpc/#api-v0-version
+                    Map res = new HashMap<>();
+                    res.put("Version", "0.4.11");
```

Comment:
Why this version?

Reply:
That is the version that the java ipfs client checks for as part of establishing connection. Happy to change it

Reply:
I would put in our version here

Reply:
done